taf/oll/repositoriesdb.py

```diff
@@ -0,0 +1,176 @@
+    _repositories_dict[auth_repo.name] = {}
+  elif commit in _repositories_dict[auth_repo.name]:
+    return
+
+  if commit is None:
+    commit = auth_repo.head_commit_sha()
+  if root_dir is None:
+    root_dir = Path(auth_repo.repo_path).parent
+
+  repositories_dict = {}
+  _repositories_dict[auth_repo.name][commit] = repositories_dict
+
+  targets = auth_repo.get_json(commit, targets_path)
+  repositories = auth_repo.get_json(commit, repositories_path)
+
+  # target repositories are defined in both mirrors.json and targets.json
+  repositories = repositories['repositories']
+  targets = targets['signed']['targets']
+  for path, repo_data in repositories.items():
+    urls = repo_data['urls']
+    target = targets.get(path)
+    if target is None and only_load_targets:
+      continue
+    additional_info = _get_custom_data(repo_data, targets[path])
+
+    if factory is not None:
+      git_repo = factory(root_dir, path, urls, additional_info)
+    else:
+      git_repo_class = _determine_repo_class(repo_classes, path)
+      git_repo = git_repo_class(root_dir, path, urls, additional_info)
+
+    if not isinstance(git_repo, GitRepository):
+      raise Exception(f'{type(git_repo)} is not a subclass of GitRepository')
+
+    repositories_dict[path] = git_repo
+
+
+def _determine_repo_class(repo_classes, path):
+  # if no class is specified, return the default one
+  if repo_classes is None:
+    return GitRepository
+
+  # if only one value is specified, that means that all target repositories
+  # should be of the same class
+  if not isinstance(repo_classes, dict):
+    return repo_classes
+
+  if path in repo_classes:
+    return repo_classes[path]
+
+  if 'default' in repo_classes:
+    return repo_classes['default']
+
+  return GitRepository
+
+
+def _get_custom_data(repo, target):
+  custom = repo.get('custom', {})
+  target_custom = target.get('custom') if target is not None else None
+  if target_custom is not None:
+    custom.update(target_custom)
+  return custom
+
+
+def get_repositories_paths_by_custom_data(auth_repo, commit=None, **custom):
+  if not commit:
+    commit = auth_repo.head_commit_sha()
+
+  targets = auth_repo.get_json(commit, targets_path)
+  repositories = auth_repo.get_json(commit, repositories_path)
+  repositories = repositories['repositories']
+  tatgets = targets['signed']['targets']
+
+  def _compare(path):
+    # Check if `custom` dict is subset of targets[path]['custom'] dict
+    try:
+      return custom.items() <= _get_custom_data(repositories[path],
+                                                targets.get(path)).items()
+    except (AttributeError, KeyError):
+      return False
+
+  paths = list(filter(_compare, repositories)) if custom else list(repositories)
+  if len(paths):
+    return paths
+  raise RepositoriesNotFound(f'Repositories associated with custom data {custom} not found')
+
+
+def get_repository(auth_repo, path, commit=None):
+  return get_repositories(auth_repo, commit)['path']
```

Comment:
I guess it should be `return get_repositories(auth_repo, commit)[path]`.

Reply:
Yes